odoo/tools/profiler.py

```diff
@@ -272,6 +272,94 @@ def post_process(self):
         super().post_process()
 
 
+class QwebCollector(Collector):
+    """
+    Record qweb execution with directive trace.
+    """
+    name = 'qweb'
+
+    def __init__(self):
+        super().__init__()
+        self.archs = {}
+        self.results = []
+        self.stack = []
+
+        def hook(event, env, **kwargs):
+            if event == 'begin':
+                self._handle_begin(env, **kwargs)
+            elif event == 'end':
+                self._handle_end(env, **kwargs)
```

Comment:
I would suggest to do as less processing as possible here: profiler entries should be as fast as possible.


```suggestion
        def hook(event, env, **kwargs):
            self.result.append(event, kwargs, env.cr.sql_log_count, time.time())
```

Then you can process all entries in postprocess (supposing that kwargs are immutable)